lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -276,7 +276,13 @@ def main():
     # find the group if present
     group = None
     groups = {}
-    for curGroup in ec2.get_all_security_groups():
+
+    try:
+        security_groups = ec2.get_all_security_groups()
+    except Exception as e:
```

Comment:
Sorry about that, alexlo03. Had my head in the uh clouds. :/ Not a request for boto3! I'd prefer to see the bugfixes make it into 2.3 and fix make the exceptions better in another PR if willthames's suggestion above doesn't catch timeouts.